analyses/molecular-subtyping-HGG/09-HGG-with-braf-clustering.Rmd

```diff
@@ -0,0 +1,318 @@
+  #   plot1: the first dimension reduction plot
+  #   plot2: the second dimendion reduction plot
+  #   output_filename: the filename for the output multipanel plot
+  #
+  # Returns:
+  #   final_plot: multipanel plot with visualizations of t-SNE and UMAP scores
+  
+  # Putting the title together
+  plot_title <-
+    toupper(paste(sub(".pdf", "", output_filename)))
+
+  # Put the plots in a list
+  plot_list <- list(plot1, plot2)
+  
+  # Extract legend from the plot list + adjust text size
+  plot_legend <- cowplot::get_legend(
+    plot_list[[1]] +
+      guides(color = guide_legend(ncol = 1)) +
+      theme(
+        text = element_text(size = 10),
+        legend.box.margin = margin(15, 15, 15, 15)
+      )
+  )
+  
+  # We'll remove the legends from all the plots for arranging in a grid
+  plot_list <- lapply(plot_list,
+                      function(p)
+                        p + theme(legend.position = "none"))
+  
+  # Multipanel plot of the dimension reduction scatter plots with panel labels
+  multipanel_plot <-
+    cowplot::plot_grid(
+      plotlist = plot_list,
+      nrow = 1,
+      align = "h",
+      labels = "AUTO"
+    )
+  
+  # This is the plot title
+  plot_title_panel <- cowplot::ggdraw() +
+    cowplot::draw_label(plot_title, fontface = "bold", size = 15)
+  
+  # Add the title such that it is centered over the scatter plots
+  multipanel_with_title <-
+    cowplot::plot_grid(
+      plot_title_panel,
+      multipanel_plot,
+      ncol = 1,
+      rel_heights = c(0.1, 1)
+    )
+  
+  # Pad the right side of the multipanel plot that now has a title
+  multipanel_spacer <- cowplot::plot_grid(multipanel_with_title,
+                                          NULL,
+                                          nrow = 1,
+                                          rel_widths = c(1, 0.05))
+  
+  # Add in the legend to the right of the multipanel plot
+  final_plot <- cowplot::plot_grid(
+    multipanel_with_title,
+    plot_legend,
+    nrow = 1,
+    rel_widths = c(1, 0.25)
+  ) +
+    theme(plot.margin = unit(c(1, 1, 1, 1), "cm"))
+  
+  # Save to file
+  cowplot::save_plot(
+    file.path(plots_dir, output_filename),
+    final_plot,
+    base_width = 24,
+    base_height = 8
+  )
+  
+  # Display final plot
+  final_plot
+}
+
+```
+
+### Read in data
+
+```{r}
+# Read in HGG molecular subtype tsv file -- output of `07-HGG-molecular-subtyping-combine-table.Rmd`
+hgg_molecular_subtype_df <- readr::read_tsv(file.path("results",
+                                                      "HGG_molecular_subtype.tsv"))
+
+# Read in metadata
+metadata_df <-
+  readr::read_tsv(
+    file.path(data_dir, "pbta-histologies.tsv"),
+    col_types = cols(molecular_subtype = col_character())
+  )
+
+# Read in expression files
+stranded_tsne_aligned_scores <- readr::read_tsv(
+  file.path(
+    "..",
+    "..",
+    "analyses",
+    "transcriptomic-dimension-reduction",
+    "results",
+    "rsem_stranded_log_tsne_scores_aligned.tsv"
```

Comment:
These aligned metadata TSV files will contain the `molecular_subtype` column from `pbta-histologies.tsv` , but you will have to use `col_types` the same way you do above:

```r
    col_types = cols(molecular_subtype = col_character())
```
If these aligned metadata files are out of date because they were run with v13 of the data and not v14 which will contain the values that are of interest here (e.g., the HGG labels), let's open a new pull request + rerun `transcriptomic-dimension-reduction`. Then you can use the aligned scores TSV here and not have to do any additional "alignment" of metadata.

Reply:
The "alignment" here is done using the `HGG_molecular_subtype.tsv` file produced in this module. Does your comment here mean that the values in the `molecular_subtype` column of this file should also be in the aligned metadata files from the `transcriptomic-dimension-reduction` module?

Reply:
If you use the v14 files to run `transcriptomic-dimension-reduction` (taking care to make sure you specify the `col_type` for molecular subtype), you don't need to use `HGG_molecular_subtype.tsv` because that is just duplicate information (e.g., the `molecular_subtype` label in `pbta-histologies.tsv` _comes from_ `HGG_molecular_subtype.tsv`)